src/utilities/multicall.ts

```diff
@@ -0,0 +1,60 @@
+import { Fragment, Interface, JsonFragment } from '@ethersproject/abi';
+import { CallOverrides, Contract } from '@ethersproject/contracts';
+import { JsonRpcProvider } from '@ethersproject/providers';
+
+import { MultiCallAddress } from '../config';
+import { SupportedChainId } from '../types';
+
+export interface Call {
+  address: string; // Address of the contract
+  name: string; // Function name on the contract (example: balanceOf)
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  params?: any[]; // Function params
+}
+
+export interface MulticallOptions extends CallOverrides {
+  limit?: number; // multicall pagination
+}
+
+export const multicall = async <T = any>(
```

Comment:
I'm realizing that this isn't actually used anywhere yet. This also feels like something that might be an npm package somewhere which we should just import. If it's not already a package we should make it one.